tools/alara_output_processing/alara_output_processing.py

```diff
@@ -130,193 +290,161 @@ def extract_tables(self):
+            cooling time. If multiple runs, blocks, or variables are present
+            in the ALARADFrame, a ValueError will be raised with directions
+            for which column to further filter.
+
         Arguments:
-            self (alara_output_processing.ALARADFrame): Specialized ALARA
-                output DataFrame containing the extracted tabular data for a
-                single variable and interval/zone of an ALARA run.
-            relative (bool, optional): Option for ALARADFrames already
-                processed by relative_contributions().
-                (Defaults to False)     
-            threshold (float or int): Proportional threshold value for
-                inclusion cutoff.
-                (Defaults to 0.05)
+            self (alara_output_processing.ALARADFrame): Pre-filtered
+                ALARADFrame containing data for a single response, block, and
+                run.
+
         Returns:
-            mask_df (alara_output_processing.ALARADFrame): Processed
-                ALARADFrame (potentially) with new row, "Other", containing
-                all aggregrated data below the thresholds.
+            adf_rel (alara_output_processing.ALARADFrame): Copy of self, with
+                a new column 'rel_value' representing the relative
+                contribution of each row's value to the corresponding total at
+                each cooling time.
         '''
 
-        cols = self.columns[1:]
-        rel_adjustment = (
-            [1] * len(cols) if relative else self.extract_totals()
-            )
-        threshold_vals = {
-            col: threshold * adj for col, adj in zip(cols, rel_adjustment)
-        }
-
-        small_mask = (self[cols] < pd.Series(threshold_vals)).all(axis=1)
-        other_row = pd.Series(0.0, index=self.columns, dtype=object)
-        other_row[self.columns[0]] = 'Other'
-        other_row[cols] = self.loc[small_mask, cols].sum()
-
-        mask_adf = self.loc[~small_mask].reset_index(drop=True)
-        if other_row[cols].sum() > 0:
-            mask_adf = pd.concat(
-                [mask_adf, pd.DataFrame([other_row])], ignore_index=True
-            )
+        necessary_single_filters = ['run_lbl', 'block', 'variable']
+        for col in necessary_single_filters:
+            if len(self[col].unique()) > 1:
+                raise ValueError(
+                    'Relative values can only be calculated for single run, '\
+                    f'block, and variable ALARADFrames. \nFilter {col} to '\
+                    'single value before calculating relative values.'
+                )
 
-        return mask_adf
+        totals = self.filter_rows({'nuclide' : 'total'})['value'].to_numpy()
+        times = sorted(set(self['time']))
+        total_map = dict(zip(times, totals))
+        adf_rel = self[self['nuclide'] != 'total'].copy()
 
+        adf_rel['rel_value'] = adf_rel.apply(
```

Comment:
Since you've made a copy, I think it's OK to just change the `value` and return this as a separate DF with only the relative values.